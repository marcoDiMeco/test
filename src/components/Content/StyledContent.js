import styled from 'styled-components';

const StyledContainer = styled.div`
  background-color: #f4f4f4;
  width: 100%;
  display: flex;
  flex: 1;
`;

const StyledContent = styled.div`
  width: 100%;
  max-width: 1024px;
  margin: 0 auto;
  padding-top: 120px;
  padding-bottom: 120px;
  padding-left: 20px;
  padding-right: 20px;
  overflow: auto;
`;

export {
  StyledContainer,
  StyledContent,
};

