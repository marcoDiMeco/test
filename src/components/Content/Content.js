import React from "react";
import PropTypes from "prop-types";
import { Switch, Route, Redirect } from "react-router-dom";
import { StyledContainer, StyledContent } from "./StyledContent";
import { UsersList } from "../";

const Content = ({ users, getMoreUsers }) => {
  const { data, isLoading, isPaginationLoading, error } = users;

  return (
    <StyledContainer>
      <StyledContent>
        <Switch>
          <Route
            exact
            path="/list1"
            component={() => (
              <UsersList
                data={data}
                isLoading={isLoading}
                list={"first"}
                error={error}
              />
            )}
          />
          <Route
            exact
            path="/list2"
            component={() => (
              <UsersList
                data={data}
                isLoading={isLoading}
                isPaginationLoading={isPaginationLoading}
                getMoreUsers={getMoreUsers}
                list={"second"}
                error={error}
              />
            )}
          />
          <Redirect exact from="*" to={"/list1"} />
        </Switch>
      </StyledContent>
    </StyledContainer>
  );
};

Content.propTypes = {
  users: PropTypes.object,
  getMoreUsers: PropTypes.func,
  path: PropTypes.string,
};

Content.defaultProps = {
  users: {},
  path: "/list1",
};

export default Content;
