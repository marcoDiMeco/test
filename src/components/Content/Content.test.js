import React from "react";
import renderer from "react-test-renderer";
import { createBrowserHistory } from "history";
import { Router } from "react-router-dom";
import Content from "./";

const history = createBrowserHistory();

describe("Content component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Router history={history}>
          <Content />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

