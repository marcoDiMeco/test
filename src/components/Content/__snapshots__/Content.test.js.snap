// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content component renders correctly 1`] = `
<div
  className="sc-AxjAm hlQUYJ"
>
  <div
    className="sc-AxirZ dxHEKL"
  >
    <div
      className="sc-AxgMl gdkyin"
    />
  </div>
</div>
`;
